src/peerbook/libp2p_peerbook.erl

```diff
@@ -450,6 +476,33 @@ notify_new_peers(NewPeers, State=#state{notify_timer=NotifyTimer, notify_time=No
                      end,
     State#state{notify_peers=NewNotifyPeers, notify_timer=NewNotifyTimer}.
 
+-spec has_useful_listen_addrs(libp2p_peer:peer(), #state{}) -> boolean().
+has_useful_listen_addrs(Peer, #state{tid=TID}) ->
+    Opts = libp2p_swarm:opts(TID),
+    AllowRFC1918 = libp2p_config:get_opt(Opts, [?MODULE, allow_rfc1918], ?DEFAULT_PEERBOOK_ALLOW_RFC1918),
+    ListenAddrs0 = libp2p_peer:listen_addrs(Peer),
+    ListenAddrs = case AllowRFC1918 of
+                      true ->
+                          %% everything is ok
+                          ListenAddrs0;
+                      false ->
+                          filter_rfc1918_addresses(ListenAddrs0)
+                  end,
+    ListenAddrs /= [].
+
+filter_rfc1918_addresses(ListenAddrs) ->
```

Comment:
Maybe move part of this to transport_tcp (taking a p2p adds) so that you can use the transport_tcp function as filter instead of doing it here? Basically have another head to rfc1918